dspace-server-webapp/src/test/java/org/dspace/app/rest/EPersonRestRepositoryIT.java

```diff
@@ -1773,4 +1826,668 @@ public void getDirectEpersonGroups() throws Exception {
+                                              .andExpect(status().isCreated())
+                                              .andExpect(jsonPath("$", Matchers.allOf(
+                                                  hasJsonPath("$.uuid", not(empty())),
+                                                  // is it what you expect? EPerson.getName() returns the email...
+                                                  //hasJsonPath("$.name", is("Doe John")),
+                                                  hasJsonPath("$.email", is(newRegisterEmail)),
+                                                  hasJsonPath("$.type", is("eperson")),
+                                                  hasJsonPath("$._links.self.href", not(empty())),
+                                                  hasJsonPath("$.metadata", Matchers.allOf(
+                                                      matchMetadata("eperson.firstname", "John"),
+                                                      matchMetadata("eperson.lastname", "Doe")
+                                                  ))))).andReturn();
+
+        String content = mvcResult.getResponse().getContentAsString();
+        Map<String,Object> map = mapper.readValue(content, Map.class);
+        String epersonUuid = String.valueOf(map.get("uuid"));
+        EPerson createdEPerson = ePersonService.find(context, UUID.fromString(epersonUuid));
+        assertTrue(ePersonService.checkPassword(context, createdEPerson, "somePassword"));
+        assertNull(registrationDataService.findByToken(context, newRegisterToken));
+
+        context.turnOffAuthorisationSystem();
+        ePersonService.delete(context, createdEPerson);
+        context.restoreAuthSystemState();
+
+        context.turnOffAuthorisationSystem();
+        registrationDataService.deleteByToken(context, newRegisterToken);
+        context.restoreAuthSystemState();
+
+    }
+
+    @Test
+    public void postEPersonWithTokenWithEmailAndSelfRegisteredProperty() throws Exception {
+
+        ObjectMapper mapper = new ObjectMapper();
+
+        String newRegisterEmail = "new-register@fake-email.com";
+        RegistrationRest registrationRest = new RegistrationRest();
+        registrationRest.setEmail(newRegisterEmail);
+        getClient().perform(post("/api/eperson/registrations")
+                                .contentType(MediaType.APPLICATION_JSON)
+                                .content(mapper.writeValueAsBytes(registrationRest)))
+                   .andExpect(status().isCreated());
+        String newRegisterToken = registrationDataService.findByEmail(context, newRegisterEmail).getToken();
+
+        String json = "{\"metadata\":{\"eperson.firstname\":[{\"value\":\"John\"}]," +
+            "\"eperson.lastname\":[{\"value\":\"Doe\"}]},\"selfRegistered\":true,\"email\":\"" + newRegisterEmail +
+            "\",\"password\":\"somePassword\",\"type\":\"eperson\"}";
```

Comment:
Same in this method.  No manual JSON & correct the POST cleanup

Reply:
The reason why this is necessary for this test has been documented at https://github.com/DSpace/DSpace/pull/2763/commits/4ea4f9112948d28858fb9b8f727da75fe4312032#diff-a256d3cb30e4b4ad847136e8ffedb15aR2236

Reply:
@benbosman : I don't understand this explanation for the manually created JSON.  The inline comment you linked to says:
`When using the mapper to write an EPersonRest object to JSON to pass it along, the password gets lost`

Does this imply there's a *bug* in the how the Mapper is generating JSON from EPersonRest object?  Or maybe we need to have a different Mapper for our test environment which doesn't drop the password (if it's doing it on purpose for some reason).  I *really don't like* manually created JSON as it is very prone to typos that can cause major errors.  It's also very hard to read, as you cannot tell which brackets align with other brackets.

So, I'd like us to look more at why the mapper isn't functioning properly here and determine if maybe we need to fix it, or create a separate Mapper specifically for these tests.  Therefore, I'm going to unresolve this conversation for now.

Reply:
This is due to https://github.com/atmire/DSpace/blob/w2p-70273_create-new-registration/dspace-server-webapp/src/main/java/org/dspace/app/rest/model/EPersonRest.java#L46
It's WRITE_ONLY, so the IT will not include the password if you attach it to the EPersonRest object
Should we add a link to that line to explain this in detail?

Reply:
@benbosman : Oh, I see what you mean now. In that case, I'd recommend we override this behavior *just in our tests*.  Here's a way to do that: https://stackoverflow.com/a/55064740

Essentially, we can create that tiny `JacksonAnnotationInstrospector` class in our test classes. Then in this method, we should be able to still use a mapper, but call `mapper.setAnnotationIntrospector(new IgnoreJacksonWriteOnlyAccess());`

You could always add a brief inline comment before that line to say "We need to ignore WRITE_ONLY annotation on password proper in order to use the mapper"

Reply:
@tdonohue thanks for the suggestion, Raf has committed this for all parts where JSON was manually used in https://github.com/DSpace/DSpace/pull/2763/commits/3aba1b7545546a068f311adf7f71695699df84d9